tests/integration/rerender-test.js

```diff
@@ -0,0 +1,147 @@
+import { setComponentTemplate } from '@ember/component';
+import templateOnly from '@ember/component/template-only';
+import { precompileTemplate } from '@ember/template-compilation';
+import {
+  find,
+  getSettledState,
+  render,
+  rerender,
+  setupContext,
+  setupRenderingContext,
+  teardownContext,
+  waitFor,
+} from '@ember/test-helpers';
+import { buildWaiter } from '@ember/test-waiters';
+import GlimmerComponent from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { hbs } from 'ember-cli-htmlbars';
+import { module, test } from 'qunit';
+
+module('rerender real-world', function (hooks) {
```

Comment:
I think we need to guard this test with a short circuit return when on Ember < 3.25

```suggestion
module('rerender real-world', function (hooks) {
  // this test requires Ember 3.28 in order to use lexically scoped values in templates
  if (!hasEmberVersion(3, 28)) {
    return;
  }
```